articles/recommender-system-explained/index.md

```diff
@@ -0,0 +1,89 @@
+  
+### Components of a Recommender system
+- **Candidate generations:** This component decomposes the huge pool of candidates (potential recommendations) into smaller subsets. It then recommends these small subsets to the online user. 
+- **Scoring systems:** The scoring system allocates a score to all the items or elements in the subsets. 
+- **Re-ranking systems:** These systems consider the item scores and other constraints to generate the final rankings.
+  
+### Types of recommender systems
+The following are the main types of recommender systems:
+
+#### Collaborative filtering
+This type of recommender system uses the rating profile of users to generate recommendations. It also uses the feedback and reactions of past users to suggest products to other users. 
+
+Collaborative filtering consists of two approaches: the memory-based approach and the model-based approach. The memory-based approach suggests items based on similar user interaction profiles or similar items that the user has interacted with. The model-based approach uses information relating to the user-item interactions to generate a latent model for recommendation.
+
+#### Content-based filtering
+This type of recommender system uses the additional content of items and users to generate suggestions for items. For example, a recommender system for a movie may analyze additional information such as the main actors and the duration of the movie. 
+
+This system can also analyze information such as the sex or age of the users that watched the movie. This collective information will be used to suggest movies based on the profile of the user (sex or age).
+
+#### Multi-criteria recommender systems
+These systems provide recommendations based on multiple criteria relating to preference information. The system analyzes the preference information of a user for an explored item using various criteria, rather than a single criterion. It then forecasts a rank for unexplored products and provides relevant suggestions to the user. 
+
+#### Mobile recommender systems
+These systems use smartphone devices to offer personalized recommendations. The accuracy of these systems depends on the context, privacy, and recommendation approach. A good example of a mobile recommender system is the one used by taxi companies such as Lyft, DiDi, and Uber. In such a case, the system uses the GPS information of routes used by drivers to recommend appropriate pick-up points. The GPS information used includes the operational status, location, and time stamps. 
+
+#### Hybrid recommender systems
+Hybrid recommender systems combine content-based filtering and collaborative filtering approaches. These systems help in overcoming some of the limitations of pure recommender systems such as sparsity problems and cold start. Netflix is an example of a company that employs hybrid recommender systems.
+
```

Comment:
Briefly explain these problems or provide a hyperlink for readers to understand.